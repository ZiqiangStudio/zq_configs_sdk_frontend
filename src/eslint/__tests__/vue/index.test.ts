import { curryExecEslint } from '../util';

describe('Vue Lint', () => {
  const execEslint = curryExecEslint('vue');

  test('报错', async () => {
    const { output, error } = await execEslint('error.vue');
    expect(error).toBeTruthy();
    expect(output).toMatchSnapshot();
  });

  test('正常通过', async () => {
    const { output, error } = await execEslint('normal.vue');
    expect(error).toBeFalsy();
    expect(output).toMatchSnapshot();
  });
});
