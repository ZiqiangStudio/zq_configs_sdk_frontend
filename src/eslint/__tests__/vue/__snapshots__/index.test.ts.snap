// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue Lint 报错 1`] = `
"[0m[0m
[0m[4m/Users/bytedance/dev/zq_configs_sdk_frontend/src/eslint/__tests__/vue/error.vue[24m[0m
[0m   [2m2:1[22m   [31merror[39m  Insert \`··\`                                                                [2mprettier/prettier[22m[0m
[0m   [2m3:1[22m   [31merror[39m  Replace \`··<p·class="title">{title}·\` with \`····<p·class="title">{title}\`  [2mprettier/prettier[22m[0m
[0m   [2m4:1[22m   [31merror[39m  Insert \`··\`                                                                [2mprettier/prettier[22m[0m
[0m   [2m7:1[22m   [31merror[39m  Delete \`··\`                                                                [2mprettier/prettier[22m[0m
[0m   [2m7:3[22m   [31merror[39m  'onMount' is not defined                                                   [2mno-undef[22m[0m
[0m  [2m13:1[22m   [31merror[39m  Delete \`⏎\`                                                                 [2mprettier/prettier[22m[0m
[0m  [2m17:18[22m  [31merror[39m  Insert \`;\`                                                                 [2mprettier/prettier[22m[0m
[0m[0m
[0m[31m[1m✖ 7 problems (7 errors, 0 warnings)[22m[39m[0m
[0m[31m[1m[22m[39m[31m[1m  6 errors and 0 warnings potentially fixable with the \`--fix\` option.[22m[39m[0m
[0m[31m[1m[22m[39m[0m
"
`;

exports[`Vue Lint 正常通过 1`] = `""`;
